server/src/graql/internal/reasoner/atom/predicate/NeqIdPredicate.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * GRAKN.AI - THE KNOWLEDGE GRAPH
+ * Copyright (C) 2018 Grakn Labs Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.core.graql.internal.reasoner.atom.predicate;
+
+import com.google.common.collect.Iterables;
+import grakn.core.graql.answer.ConceptMap;
+import grakn.core.graql.internal.reasoner.atom.Atomic;
+import grakn.core.graql.internal.reasoner.query.ReasonerQuery;
+import grakn.core.graql.query.pattern.property.NeqProperty;
+import grakn.core.graql.query.pattern.statement.Statement;
+import grakn.core.graql.query.pattern.statement.Variable;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * <p>
+ * Predicate implementation specialising it to be an inequality predicate. Corresponds to graql {@link NeqProperty}.
+ * </p>
+ *
+ *
+ */
+
+public class NeqIdPredicate extends NeqPredicate {
+
+    private NeqIdPredicate(Variable varName, Variable predicateVar, Statement pattern, ReasonerQuery parentQuery) {
+        super(varName, predicateVar, pattern, parentQuery);
+    }
+
+    public static NeqIdPredicate create(Statement pattern, ReasonerQuery parent) {
+        return new NeqIdPredicate(pattern.var(), extractPredicateVariable(pattern), pattern, parent);
+    }
+    public static NeqIdPredicate create(Variable varName, NeqProperty prop, ReasonerQuery parent) {
+        Statement pattern = new Statement(varName).not(prop);
+        return create(pattern, parent);
+    }
+
+    private static Variable extractPredicateVariable(Statement pattern) {
+        return Iterables.getOnlyElement(pattern.getProperties(NeqProperty.class).collect(Collectors.toSet())).statement().var();
+    }
+
+    @Override
+    public Atomic copy(ReasonerQuery parent) { return create(this.getPattern(), parent);}
+
+    @Override
+    public String toString(){
+        IdPredicate idPredicate = this.getIdPredicate(this.getVarName());
+        IdPredicate refIdPredicate = this.getIdPredicate(this.getPredicate());
+        return "[" + getVarName() + "!=" + getPredicate() + "]" +
+                (idPredicate != null? idPredicate : "" ) +
+                (refIdPredicate != null? refIdPredicate : "");
+    }
```

Comment:
Can we have syntax-creation like this be implemented using the `Query` syntax enums, Kasper? These things will change in the future and we want to make sure we manage them consistently. (In fact, I'm about to make a change to the inequality operators syntax in the next PR).

Reply:
It's for output and printing when debugging only

Reply:
Still, though. It's not just "good practice" to always manage string values in constant variables, but also, in this case, is required so we can ensure syntax updates are synchronised across the codebase, even for debug statements. Otherwise, the debug statements would become misleading. @kasper-piskorski

Reply:
the new syntax enums are at `graql.util.Token` bw (which will moved to `graql.lang.util.Token` in the next PR)